stage2/client.py

```diff
@@ -146,20 +146,56 @@ def udp_receive_messages(self, udp_socket):
                 print(
                     f"Key error: {e}. The received message does not have the expected format."
                 )
+                
+    def prompt_and_validate_username(self) -> str:
+        max_bytes_in_username = 255
+        
+        while True:
+            username = input("Enter your username: ")
+            
+            if len(username.encode()) > max_bytes_in_username:
+                print("The username exceeds the maximum character limit.")
+                
+            elif len(username) <= 0:
+                print("Username is required.")
+                
+            else:
+                return username
```

Comment:
大した点ではないのですが、変数user_nameで統一したいです。
```suggestion
    def prompt_and_validate_user_name(self) -> str:
        max_bytes_in_user_name = 255
        
        while True:
            user_name = input("Enter your username: ")
            
            if len(user_name.encode()) > max_bytes_in_user_name:
                print("The username exceeds the maximum character limit.")
                
            elif len(user_name) <= 0:
                print("Username is required.")
                
            else:
                return user_name
```